feature-libs/my-account/organization/components/unit/services/unit-list.service.ts

```diff
@@ -17,60 +25,118 @@ import { OrganizationTableType } from '../../shared/organization.model';
 export class UnitListService extends OrganizationListService<B2BUnit> {
   protected tableType = OrganizationTableType.UNIT;
 
+  protected initialExpanded = 1;
+
+  protected expandState = new Map<string, boolean>();
+  protected levelState = new Map<string, number>();
+  protected update$ = new BehaviorSubject(undefined);
+
+  protected expandedNodeMap = {};
   constructor(
     protected tableService: TableService,
-    protected unitService: OrgUnitService
+    protected unitService: OrgUnitService,
+    protected unitItemService: UnitItemService
   ) {
     super(tableService);
+    this.waitForKey();
   }
 
-  protected load(): Observable<EntitiesModel<B2BUnit>> {
-    return this.unitService
+  protected waitForKey() {
+    this.unitService
       .getTree()
-      .pipe(map((raw) => this.convertUnits(raw)));
+      .pipe(
+        withLatestFrom(this.unitItemService.key$),
+        first(([_tree, id]) => Boolean(id))
+      )
+      .subscribe(([tree, id]) => {
+        this.expandBranch(id, tree);
+        this.update$.next(true);
+      });
+  }
+
+  protected load(): Observable<EntitiesModel<B2BUnit>> {
+    return this.update$.pipe(
+      switchMap(() =>
+        this.unitService
+          .getTree()
+          .pipe(map((tree: B2BUnitNode) => this.convertListItem(tree)))
+      )
+    );
+  }
+
+  protected convertListItem(
+    unit: B2BUnitNode,
+    depthLevel = 0,
+    pagination = { totalResults: 0 }
+  ): EntitiesModel<B2BUnit> {
+    let values = [];
+    const expanded = this.isExpanded(unit.id, depthLevel);
+    const treeNode: B2bUnitTreeNode = {
+      ...unit,
+      count: unit.children?.length ?? 0,
+      expanded: expanded,
+      depthLevel,
+      // tmp, should be normalized
+      uid: unit.id,
+    };
+    this.expandState.set(treeNode.uid, expanded);
+    this.levelState.set(treeNode.uid, depthLevel);
+
+    values.push(treeNode);
+    pagination.totalResults++;
```

Comment:
I was under the assumption that we don't have pagination in the tree component, or am I wrong?

Reply:
Yes, we don't have pagination here, but 'pagination' object is used in list header, I decided for wrapping it to be compatible with current components. (simulate / workaround backend output)